subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -76,12 +77,12 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 struct observe_node {
 	sys_snode_t node;
-	struct lwm2m_obj_path path;
-	uint8_t  token[MAX_TOKEN_LEN];
-	int64_t event_timestamp;
-	int64_t last_timestamp;
-	uint32_t min_period_sec;
-	uint32_t max_period_sec;
+	sys_slist_t path_list;		/* List of Observation path */
+	bool composite :1; 		/* Composite Observation */
```

Comment:
I don't see a reason for using bit fields.

Reply:
We need to know Observe node role in couple places is it composite or single path node.

Reply:
If we have 2 boolean without bit field it allocate 2 byte now it allocate just 1 byte

Reply:
That is true for the RAM usage. But when you introduce bit fields, you tend to increase the code size and add some delay as the compiler needs to provide some extra code when using and assigning these bit fielded values, therefore one could argue against this bitfielding. The RAM usage with default configuration currently would take 15 bytes more RAM if the booleans weren't bitfielded as the structs are defined as static.

Reply:
That is true for the RAM usage. But when you introduce bit fields, you tend to increase the code size and add some delay as the compiler needs to provide some extra code when using and assigning these bit fielded values, therefore one could argue against this bitfielding. The RAM usage with default configuration currently would take 15 bytes more RAM if the booleans weren't bitfielded as the structs are defined as static.